### PR TITLE
Added API v2 Checkout traversing endpoints

### PR DESCRIPTION
**Pull request by damianlegawiec**

### CODE REVIEW

File: api/docs/v2/storefront/index.yaml [hunk 1]
```diff
@@ -285,6 +357,9 @@ components:
           type: number
           example: 10
           description: 'Numnber of all pages containing items matching the criteria'
```
Comment (VygovskySergey) [inline api/docs/v2/storefront/index.yaml]: `Number`

File: api/docs/v2/storefront/index.yaml [hunk 2]
```diff
@@ -146,6 +146,78 @@ paths:
...
+                $ref: '#/components/schemas/Error'
+        '404':
+          $ref: '#/components/responses/404NotFound'
+      security:
+        - orderToken: []
+        - bearerAuth: []
+
+  '/checkout/advance':
+    patch:
+      description: Advances Checkout to the farthest Checkout step validation allows, until the Complete step
```
Comment (kushniryb) [inline api/docs/v2/storefront/index.yaml]: `furthest`

File: api/docs/v2/storefront/index.yaml [hunk 3]
```diff
@@ -307,32 +382,71 @@ components:
...
+                display_tax_total:
+                  type: string
+                  example: '$10.00'
+                total_quantity:
+                  type: number
+                  example: 2
+                  description: 'Total quantity number of all items added to the Cart'
+                special_instructions:
+                  type: string
+                  example: 'Please ship ASAP!!!111'
```
Comment (kushniryb) [inline api/docs/v2/storefront/index.yaml]: Can we provide more solid example here? 😄

Comment (damianlegawiec) [reply api/docs/v2/storefront/index.yaml]: @kushniryb why so serious? ;)

**Review by WitekL — CHANGES_REQUESTED** [review]

**Comment by damianlegawiec** [comment]
@WitekL fixed!

**Review by WitekL — APPROVED** [review]
👍

**Review by VygovskySergey — APPROVED** [review]

**Review by kushniryb — APPROVED** [review]

**Pull request merged**